flytepropeller/pkg/controller/nodes/attr_path_resolver.go

```diff
@@ -97,6 +105,79 @@ func resolveAttrPathInPbStruct(nodeID string, st *structpb.Struct, bindAttrPath
 	return literal, err
 }
 
+// resolveAttrPathInBinary resolves the binary idl object (e.g. dataclass, pydantic basemodel) with attribute path
+func resolveAttrPathInBinary(nodeID string, binaryIDL *core.Binary, bindAttrPath []*core.PromiseAttribute) (*core.
+	Literal,
+	error) {
+
+	binaryBytes := binaryIDL.GetValue()
+	serializationFormat := binaryIDL.GetTag()
+
+	var currVal interface{}
+	var tmpVal interface{}
+	var exist bool
+
+	if serializationFormat == messagepack {
+		err := msgpack.Unmarshal(binaryBytes, &currVal)
+		if err != nil {
+			return nil, err
+
+		}
+	} else {
+		return nil, errors.Errorf(errors.PromiseAttributeResolveError, nodeID,
+			"Unsupported format '%v' found for literal value.\n"+
+				"Please ensure the serialization format is supported.", serializationFormat)
+	}
+
+	// Turn the current value to a map, so it can be resolved more easily
+	for _, attr := range bindAttrPath {
+		switch resolvedVal := currVal.(type) {
+		// map
+		case map[interface{}]interface{}:
+			tmpVal, exist = resolvedVal[attr.GetStringValue()]
+			if !exist {
+				return nil, errors.Errorf(errors.PromiseAttributeResolveError, nodeID, "key [%v] does not exist in literal %v", attr.GetStringValue(), currVal)
+			}
+			currVal = tmpVal
+		// list
+		case []interface{}:
+			if int(attr.GetIntValue()) >= len(resolvedVal) {
+				return nil, errors.Errorf(errors.PromiseAttributeResolveError, nodeID, "index [%v] is out of range of %v", attr.GetIntValue(), currVal)
+			}
+			currVal = resolvedVal[attr.GetIntValue()]
+		}
+	}
+
+	if serializationFormat == messagepack {
+		// Marshal the current value to MessagePack bytes
+		resolvedBinaryBytes, err := msgpack.Marshal(currVal)
+		if err != nil {
+			return nil, err
+		}
+		// Construct and return the binary-encoded literal
+		return constructResolvedBinary(resolvedBinaryBytes, serializationFormat), nil
+	}
+	// Unsupported serialization format
+	return nil, errors.Errorf(errors.PromiseAttributeResolveError, nodeID,
+		"Unsupported format '%v' found for literal value.\n"+
+			"Please ensure the serialization format is supported.", serializationFormat)
```

Comment:
This case can never happen, right? since you checked for that early on

Reply:
Yes, I can remove it